### PR TITLE
kumo 1.13 (new formula)

### PR DESCRIPTION
**Pull request by kennycason**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/kumo.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Kumo < Formula
+  desc "Kumo: Word Clouds in Java"
```
Comment (fxcoudert) [inline Formula/kumo.rb]: Remove `Kumo: `

File: Formula/kumo.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  homepage "https://github.com/kennycason/kumo"
+  url "https://search.maven.org/remotecontent?filepath=com/kennycason/kumo-cli/1.12/kumo-cli-1.12.jar"
+  sha256 "9d67d943676be57fe75d47f1832c4e8cb415ee8e4e82a71d4d7d1e7b4effe7c4"
+
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install "kumo-cli-1.12.jar"
+    bin.write_jar_script libexec/"kumo-cli-1.12.jar", "kumo"
+    puts "Finished installing kumo 1.12"
```
Comment (fxcoudert) [inline Formula/kumo.rb]: Remove this line

File: Formula/kumo.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def server_script(server_jar); <<-EOS.undent
```
Comment (fxcoudert) [inline Formula/kumo.rb]: What is this function trying to achieve? It is never called…

File: Formula/kumo.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    #!/bin/bash
+    exec java -cp #{server_jar} com.kennycason.kumo.cli.KumoCli "$@"
+  EOS
+  end
+
+  test do
+    pipe_output("#{bin}/kumo --version", "Test Kumo version command")
```
Comment (fxcoudert) [inline Formula/kumo.rb]: We need a test that does more than that… like writes an actual image

**Review by fxcoudert — DISMISSED** [review]

File: Formula/kumo.rb (hunk 3 above)
Comment (kennycason) [reply Formula/kumo.rb]: Copy-pasta

Comment (kennycason) [reply Formula/kumo.rb]: This is left over from a previous example I found within homebrew. I had meant to remove it.

File: Formula/kumo.rb (hunk 4 above)
Comment (kennycason) [reply Formula/kumo.rb]: Those tests are already handled in the code base. Those seem too specific to be part of the tool installation validation. It's easy enough to add, but I question the value. Asserting the jar is accessible and responding seems sufficient.

To test that from the CLI tool, I'd have to A) load from a fixed url which I'm ok with but I'd pick something like wikipedia.org, not sure if that's ok, or B) provide a static file, which is less doable, and C) pick some safe place to output the generated image (/tmp/?).

To make the test be any more valuable than the print version, I would also need to form some sort of assert to verify the image contents of the word cloud were "correct", which isn't really doable.

Comment (kennycason) [reply Formula/kumo.rb]: If this is decided to be absolutely required, I have prepared:
```
pipe_output("#{bin}/kumo --version", "Test Kumo version command")
pipe_output("#{bin}/kumo -i https://wikipedia.org -o /tmp/wikipedia.png", "Generate simple wordcloud")
```

**Comment by kennycason** [comment]
@fxcoudert Thanks for the review! I'm making updates and will re-push shortly.

**Comment by kennycason** [comment]
@fxcoudert Updates pushed. Let me know your/homebrew's thoughts on the test section in regards to my previous comments. Thanks.

File: Formula/kumo.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install "kumo-cli-1.13.jar"
+    bin.write_jar_script libexec/"kumo-cli-1.13.jar", "kumo"
+  end
+
+  test do
+    pipe_output("#{bin}/kumo --version", "Test Kumo version command")
+    pipe_output("#{bin}/kumo -i https://wikipedia.org -o /tmp/wikipedia.png", "Generate simple wordcloud")
```
Comment (DomT4) [inline Formula/kumo.rb]: `#{testpath}` not `/tmp`.

File: Formula/kumo.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install "kumo-cli-1.13.jar"
+    bin.write_jar_script libexec/"kumo-cli-1.13.jar", "kumo"
+  end
+
+  test do
+    pipe_output("#{bin}/kumo --version", "Test Kumo version command")
```
Comment (DomT4) [inline Formula/kumo.rb]: Why the use of `pipe_output()` here?

Comment (kennycason) [reply Formula/kumo.rb]: No idea, copied from another example, there are ~250+ examples of it in codebase. I know that's not a good reason :)

Comment (kennycason) [reply Formula/kumo.rb]: I found the source code here. https://github.com/Homebrew/brew/blob/e72fde418b439b45c6036774159b8e2479faaefe/Library/Homebrew/formula_assertions.rb

I believe the example I copied from also misused pipe_output. I will update to actually assert exit codes.

File: Formula/kumo.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+class Kumo < Formula
+  desc "Word Clouds in Java"
+  homepage "https://github.com/kennycason/kumo"
+  url "https://search.maven.org/remotecontent?filepath=com/kennycason/kumo-cli/1.13/kumo-cli-1.13.jar"
+  sha256 "c9ad525f7d6aec9e2c06cf10017e1e533f43b1b3c4df5aa0d4b137f8d563c5c6"
+
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install "kumo-cli-1.13.jar"
```
Comment (DomT4) [inline Formula/kumo.rb]: Use `#{version}` rather than `1.13` here and below.

File: Formula/kumo.rb (hunk 6 above)
Comment (DomT4) [reply Formula/kumo.rb]: I think you want something like:
```ruby
system bin/"kumo", "-i", "https://wikipedia.org", "-o", testpath/"wikipedia.png"
assert_predicate testpath/"wikipedia.png", :exist?, "Wordcloud was not generated!"
```

Comment (kennycason) [reply Formula/kumo.rb]: Even better. Thanks. :) will update

**Comment by kennycason** [comment]
@DomT4 your suggestions worked like a charm. Updated and ran brew test to verify.

File: Formula/kumo.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
+class Kumo < Formula
+  desc "Word Clouds in Java"
+  homepage "https://github.com/kennycason/kumo"
+  url "https://search.maven.org/remotecontent?filepath=com/kennycason/kumo-cli/1.13/kumo-cli-1.13.jar"
+  sha256 "c9ad525f7d6aec9e2c06cf10017e1e533f43b1b3c4df5aa0d4b137f8d563c5c6"
+
```
Comment (DomT4) [inline Formula/kumo.rb]: Add a newline here and then `bottle :unneeded`.

**Comment by kennycason** [comment]
@DomT4 updated.

File: Formula/kumo.rb [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  bottle :unneeded
+
+  depends_on :java => "1.8+"
```
Comment (DomT4) [inline Formula/kumo.rb]: Just to check, does this actually work with Java 9?

**Review by DomT4 — APPROVED** [review]
It's good with me at least. CC @ilovezfs for his review.

File: Formula/kumo.rb [hunk 10]
```diff
@@ -0,0 +1,20 @@
+class Kumo < Formula
+  desc "Word Clouds in Java"
```
Comment (ilovezfs) [inline Formula/kumo.rb]: why is the C in Clouds capitalized

Comment (ilovezfs) [reply Formula/kumo.rb]: I guess upstream likes doing that :see_no_evil:

Comment (DomT4) [reply Formula/kumo.rb]: Yeah I left it because it mimics upstream 😸. It could be changed but, depends how much you care I guess.

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @kennycason! Without people like you submitting PRs we couldn't run this project. You rock!

File: Formula/kumo.rb (hunk 10 above)
Comment (ilovezfs) [reply Formula/kumo.rb]: I'm going to assume Word Clouds are now a thing like The Internet and will eventually be lower case again like the internet.

Comment (kennycason) [reply Formula/kumo.rb]: That's a very good question! haha I have never thought about it. I have spent way more of my professional career than I care to admit working with word clouds. :)

**Comment by kennycason** [comment]
@ilovezfs my pleasure. It's nice to have something as awesome as Homebrew for package management. :)

**Pull request merged**